### PR TITLE
Improve async checklist code. Follow-up to #27926

### PR DESCRIPTION
**Pull request by sgomes**
#### Changes proposed in this Pull Request

Code improvements to async checklist merged in #27926 as per the review in that PR. It was decided these could be done after merging, to reduce the time the PR stayed open and generating merge conflicts.

#### Testing instructions

Try the various checklist variations and ensure that the behaviour remains the same.

### CODE REVIEW

File: client/my-sites/checklist/wpcom-checklist/checklist-banner/index.jsx [hunk 1]
```diff
@@ -35,12 +35,6 @@ export class ChecklistBanner extends Component {
 		setNeverShowBannerStatus( siteId, true );
 
 		this.setState( { closed: true } );
-
-		if ( this.props.track ) {
-			this.props.track( 'calypso_checklist_banner_close', {
-				site_id: siteId,
-			} );
-		}
```
Comment (jsnajdr) [inline client/my-sites/checklist/wpcom-checklist/checklist-banner/index.jsx]: The original author clearly had intent to send the `calypso_checklist_banner_close` tracks event, but then something went wrong :smile: Let's make the `tracks` call work great again.

It got broken in this commit: https://github.com/Automattic/wp-calypso/commit/29af3b0fe0e2cd755a7377e207ebb0db8071722f. Part of #26764 by @ockham and @sirreal. 

That commit removes file _my-sites/stats/checklist-banner/index.jsx_ which [https://github.com/Automattic/wp-calypso/commit/29af3b0fe0e2cd755a7377e207ebb0db8071722f#diff-435657f5f9b9c272e276134ceda0644dL222](connected) the `track` prop to `recordTracksEvent` action.

And creates a new file, _my-sites/checklist/wpcom-checklist/checklist-banner/index.js_, which still calls `this.props.tracks`, but doesn't do the connect. ⚡️💥

Comment (sirreal) [reply client/my-sites/checklist/wpcom-checklist/checklist-banner/index.jsx]: Thanks for spotting this 🙇 

Will note the bad period: 623-gh-tracks-events-validation-status

Comment (jsnajdr) [reply client/my-sites/checklist/wpcom-checklist/checklist-banner/index.jsx]: Looks good :+1:

Comment (sgomes) [reply client/my-sites/checklist/wpcom-checklist/checklist-banner/index.jsx]: Thanks, @jsnajdr ! Also, by the way, excellent sleuthing work!

**Pull request merged**